xmscore/python/misc/PyUtils.cpp

```diff
@@ -168,29 +169,18 @@ py::iterable PyIterFromVecPt2d(const VecPt2d& pts)
 /// \param[in] pt: py::iterable object that represents a VecPt3d2d
 /// \return a boost::shared_ptr to a VecPt3d2d
 //------------------------------------------------------------------------------
-boost::shared_ptr<VecPt3d2d> VecPt3d2dFromPyIter(const py::iterable& pts)
+boost::shared_ptr<VecPt3d2d> VecPt3d2dFromPyIter(const py::iterable& pt3d2d)
 {
-  boost::shared_ptr<VecPt3d2d> vec_pt3d2d(new VecPt3d2d(py::len(pts)));
+  boost::shared_ptr<VecPt3d2d> vec_pt3d2d(new VecPt3d2d(py::len(pt3d2d)));
   int i = 0;
-  for (auto poly : pts) {
-      if (!py::isinstance<py::iterable>(poly)) {
-          throw py::type_error("Second arg must be an n-tuple of n-tuples of 3-tuples");
-      }
-      xms::VecPt3d vec_pt3d(py::len(poly));
-      int j = 0;
-      for (auto p : poly) {
-          if (!py::isinstance<py::iterable>(poly)) {
-              throw py::type_error("Second arg must be an n-tuple of n-tuples of 3-tuples");
-          }
-          py::tuple pt = p.cast<py::tuple>();
-          if (py::len(pt) != 3) {
-              throw py::type_error("Second arg must be an n-tuple of n-tuples of 3-tuples");
-          } else {
-              xms::Pt3d point(pt[0].cast<double>(), pt[1].cast<double>(), pt[2].cast<double>());
-              vec_pt3d.at(j) = point;
-          }
+  for (auto pts : pt3d2d) {
+      if (!py::isinstance<py::iterable>(pts)) {
+          throw py::type_error("Second arg must be an n-tuple of n-tuples of 1, 2, or 3-tuples");
```

Comment:
This error doesn’t match what we are checking. If the error message is correct, then we should adjust our checks accordingly.

Reply:
fixed